x-pack/platform/plugins/shared/task_manager/server/task_store.ts

```diff
@@ -156,6 +183,103 @@ export class TaskStore {
       maxRetries: 0,
     });
     this.requestTimeouts = opts.requestTimeouts;
+    this.security = opts.security;
+    this.spaces = opts.spaces;
+    this.canEncryptSavedObjects = opts.canEncryptSavedObjects;
+  }
+
+  public registerEncryptedSavedObjectsClient(client: EncryptedSavedObjectsClient) {
+    this.esoClient = client;
+  }
+
+  private canEncryptSo() {
+    return !!(this.esoClient && this.canEncryptSavedObjects);
+  }
+
+  private getSoClientForCreate(options: ApiKeyOptions) {
+    if (options.request) {
+      return this.savedObjectsService.getScopedClient(options.request, {
+        includedHiddenTypes: [TASK_SO_NAME],
+        excludedExtensions: [SECURITY_EXTENSION_ID, SPACES_EXTENSION_ID],
+      });
+    }
+    return this.savedObjectsRepository;
+  }
+
+  private async maybeGetApiKeyFromRequest(request?: KibanaRequest) {
+    if (!request) {
+      return null;
+    }
+
+    let userScopeAndApiKey;
+    try {
+      userScopeAndApiKey = await getApiKeyAndUserScope(
+        request,
+        this.canEncryptSo(),
+        this.security,
+        this.spaces
+      );
+    } catch (e) {
+      this.errors$.next(e);
+      throw e;
+    }
+
+    return userScopeAndApiKey;
+  }
+
+  private async bulkGetDecryptedTasks(ids: string[]) {
+    const result = new Map<string, ConcreteTaskInstance>();
+    if (!this.canEncryptSo() || !ids.length) {
+      return result;
+    }
+
+    const kueryNode = nodeBuilder.or(
+      ids.map((id) => {
+        return nodeBuilder.is(`${TASK_SO_NAME}.id`, `${TASK_SO_NAME}:${id}`);
+      })
+    );
+
+    const finder =
+      await this.esoClient!.createPointInTimeFinderDecryptedAsInternalUser<SerializedConcreteTaskInstance>(
+        {
+          type: TASK_SO_NAME,
+          filter: kueryNode,
+        }
+      );
+
+    for await (const response of finder.find()) {
+      response.saved_objects.forEach((savedObject) => {
+        result.set(savedObject.id, savedObjectToConcreteTaskInstance(savedObject));
```

Comment:
The task params and state can sometimes get pretty large. WDYT of just returning the `apiKey` and `userScope` here instead of the full task since that's all we use downstream?